alg/gdalwarpkernel.cpp

```diff
@@ -7599,13 +7603,44 @@ static void GWKAverageOrModeThread(void *pData)
 
                                     // Check array for existing entry.
                                     for (i = 0; i < iMaxInd; ++i)
-                                        if (pafRealVals[i] == fVal &&
-                                            ++panRealSums[i] >
-                                                panRealSums[iMaxVal])
+                                    {
+                                        if (pafRealVals[i] == fVal)
                                         {
-                                            iMaxVal = i;
```

Comment:
@rouault probably not worth backporting, but this is doing extra work by not hitting `break` on `pafRealVals[i] == fVal` regardless of whether `++panRealSums[i] > panRealSums[iMaxVal]`